Automattic/vip-design-system
# [PIE-3778] Improve Notice component padding and remove unused Notification component

## Description

Jira: https://vipjira.atlassian.net/browse/PIE-3778

The Notice component icon should be aligned to the title and not to the center of the content.
This PR addresses this, plus fixes a problem with the margin, in mobile view (or in smaller alerts) where the CSS Flex would shrink the margin, making them look more packed then they should've been.

Last but not least, in this PR we're removing the Notice component.

## Before and after

| Before  | After |
| ------------- | ------------- |
| ![CleanShot 2023-10-16 at 11 36 53@2x](https://github.com/Automattic/vip-design-system/assets/668251/049aae41-a676-447a-9052-db1ed1ff1e77)  | ![CleanShot 2023-10-16 at 11 35 07@2x](https://github.com/Automattic/vip-design-system/assets/668251/26ea3287-f1a2-4e40-a652-e86f726efc98)  |
| ![CleanShot 2023-10-16 at 11 42 20@2x](https://github.com/Automattic/vip-design-system/assets/668251/522fb516-936c-484a-9fb0-26386c89a98b)  | ![CleanShot 2023-10-16 at 11 42 48@2x](https://github.com/Automattic/vip-design-system/assets/668251/03eb9b37-74ec-4e4b-952c-4420a20d144b)  |


## Checklist


## Steps to Test

1. Pull down PR.
1. `npm run dev`.
1. Open Storybook.
1. Check that the `Notice` component displays as expected http://localhost:6006/?path=/story/notice--default
